homeassistant/components/asuswrt/router.py

```diff
@@ -37,14 +38,94 @@
     DEFAULT_TRACK_UNKNOWN,
     DOMAIN,
     PROTOCOL_TELNET,
+    SENSOR_CONNECTED_DEVICE,
+    SENSOR_RX_BYTES,
+    SENSOR_RX_RATES,
+    SENSOR_TX_BYTES,
+    SENSOR_TX_RATES,
 )
 
 CONF_REQ_RELOAD = [CONF_DNSMASQ, CONF_INTERFACE, CONF_REQUIRE_IP]
+
+KEY_COORDINATOR = "coordinator"
+KEY_SENSORS = "sensors"
+
 SCAN_INTERVAL = timedelta(seconds=30)
 
+SENSORS_TYPE_BYTES = "sensors_bytes"
+SENSORS_TYPE_COUNT = "sensors_count"
+SENSORS_TYPE_RATES = "sensors_rates"
+
 _LOGGER = logging.getLogger(__name__)
 
 
+class AsusWrtSensorDataHandler:
+    """Data handler for AsusWrt sensor."""
+
+    def __init__(self, hass, api):
+        """Initialize a AsusWrt sensor data handler."""
+        self._hass = hass
+        self._api = api
+        self._connected_devices = 0
+
+    async def _get_connected_devices(self):
+        """Return number of connected devices."""
+        return {SENSOR_CONNECTED_DEVICE: self._connected_devices}
+
+    async def _get_bytes(self):
+        """Fetch byte information from the router."""
+        ret_dict: Dict[str, Any] = {}
+        try:
+            datas = await self._api.async_get_bytes_total()
+        except OSError as exc:
+            raise UpdateFailed from exc
+
+        ret_dict[SENSOR_RX_BYTES] = datas[0]
+        ret_dict[SENSOR_TX_BYTES] = datas[1]
+
+        return ret_dict
+
+    async def _get_rates(self):
+        """Fetch rates information from the router."""
+        ret_dict: Dict[str, Any] = {}
+        try:
+            rates = await self._api.async_get_current_transfer_rates()
+        except OSError as exc:
+            raise UpdateFailed from exc
+
+        ret_dict[SENSOR_RX_RATES] = rates[0]
+        ret_dict[SENSOR_TX_RATES] = rates[1]
+
+        return ret_dict
+
+    def update_device_count(self, conn_devices):
+        """Update connected devices attribute."""
+        self._connected_devices = conn_devices
+
+    async def get_coordinator(self, sensor_type: str):
+        """Get the coordinator for a specific sensor type."""
+        if sensor_type == SENSORS_TYPE_COUNT:
+            method = self._get_connected_devices
```

Comment:
I just wonder if make sense create a coordinator for a sensor that do not use an API call or if it should be implemented in a different way...

Reply:
May be with last commit make more sense...